design-documents/graph-ql/coverage/mergeCarts-optional-destination.md

```diff
@@ -0,0 +1,40 @@
+# `mergeCarts` - Make `destination_cart_id` optional
```

Comment:
Please modify existing document https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/shared-cart.md instead of creating a new one.

BTW, this option was listed in the original proposal as an alternative, but was rejected because of less clear semantics.

Reply:
> BTW, this option was listed in the original proposal as an alternative, but was rejected because of less clear semantics.

I looked through the document and didn't see any mention of why this strategy was rejected. Can you help me understand what about the API was unclear? `createEmptyCart` returning a populated cart is pretty unclear as well, so I'd like to solve both problems at once if possible.

Reply:
> I would still prefer to work in scope of one document because otherwise it will be confusing. This repository should ideally contain the desired state, not two different states in different documents.

No problem - I'll move it over.

> Initially we had a requirement to merge guest carts as well. I propose to discuss it again with PO and in case the only supported case is guest cart merge into customer cart

I can do that - happen to know who the PM is?

Reply:
@nrkapoor is usually making decisions on GraphQL schema and involves someone else if needed

Reply:
Oh cool - wasn't sure if we were talking a PO than manages Ordering/Checkout or just GraphQl in general. I'll ping him

Reply:
If we end up introducing a new mutation (discussed in the other comment thread), it won't be critical for us to make changes to `mergeCarts`

Reply:
this was merged part of MC-37413  as described here. i will update https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/shared-cart.md as suggested by Alex